src/main/java/com/palash/onedrive4j/OneDrivePath.java

```diff
@@ -0,0 +1,179 @@
+package com.palash.onedrive4j;
+
+import com.nickdsantos.onedrive4j.DriveItem;
+import com.nickdsantos.onedrive4j.OneDrive;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.*;
+import java.util.Iterator;
+
+/**
+ * Created by palash on 7/29/2016.
+ */
+public class OneDrivePath implements Path {
+    private FileSystem fs;
+
+    public DriveItem getDriveItem() {
+        return driveItem;
+    }
+
+    private DriveItem driveItem;
+
+    public OneDrivePath(FileSystem fs, DriveItem driveItem) {
+        this.fs = fs;
+        this.driveItem = driveItem;
+    }
+
+    public OneDrive getOneDrive() {
+        return ((OneDriveFileSystemProvider) fs.provider()).getOneDrive();
+    }
+
+    @Override
+    public FileSystem getFileSystem() {
+        return fs;
+    }
+
+    @Override
+    public boolean isAbsolute() {
+        throw new NotImplementedException();
```

Comment:
A lot of these methods look fairly straight-forward to implement. It's a wonder there is no abstract class for implementing new Path classes...